src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsProjectJsonToPackageReferenceMigrator.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using EnvDTE;
+using NuGet.PackageManagement.UI;
+using NuGet.PackageManagement.VisualStudio;
+using System;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Threading;
+
+namespace NuGet.VisualStudio
+{
+    [Export(typeof(IVsProjectJsonToPackageReferenceMigrator))]
+    internal class VsProjectJsonToPackageReferenceMigrator : IVsProjectJsonToPackageReferenceMigrator
+    {
+        private readonly IVsSolutionManager _solutionManager;
+
+        [ImportingConstructor]
+        public VsProjectJsonToPackageReferenceMigrator(IVsSolutionManager solutionManager)
+        {
+            _solutionManager = solutionManager;
+        }
+        public IVsProjectJsonToPackageReferenceMigrateResult MigrateProjectToPackageRef(string projectUniqueName)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            if(string.IsNullOrEmpty(projectUniqueName))
+            {
+                throw new ArgumentNullException(nameof(projectUniqueName));
+            }
+
+            if(!File.Exists(projectUniqueName))
+            {
+                throw new FileNotFoundException(nameof(projectUniqueName));
+            }
+
+            return  NuGetUIThreadHelper.JoinableTaskFactory.Run(async () => 
+            {
+                await TaskScheduler.Default;
+                return await MigrateProjectToPackageRefAsync(projectUniqueName);
+            });
+            
+        }
+
+        private async Task<IVsProjectJsonToPackageReferenceMigrateResult> MigrateProjectToPackageRefAsync(string projectUniqueName)
+        {            
+            var project = _solutionManager.GetDTEProject(projectUniqueName);
+
+            if (project == null)
+            {
+                throw new InvalidOperationException(string.Format("Project {0} does not exist in the project system cache.", projectUniqueName));
+            }
+
+            var projectSafeName = await EnvDTEProjectUtility.GetCustomUniqueNameAsync(project);
+
+            var nuGetProject = _solutionManager.GetNuGetProject(projectSafeName);
+            try
+            {
+                var legacyPackageRefBasedProject = await
+                    _solutionManager.UpdateNuGetProjectToPackageRef(nuGetProject) as LegacyCSProjPackageReferenceProject;
```

Comment:
`LegacyCSProjPackageReferenceProject` is only available for Dev15. It will give compilation error for Dev14